notesapi/v1/tests/test_views.py

```diff
@@ -680,6 +681,37 @@ def search_and_verify(searchText, expectedText, expectedTags):
         search_and_verify("First", "First one", [])
         search_and_verify("Second", "Second note", ["tag1", "tag2"])
 
+    def test_bulk_search(self):
+        """
+        Tests for bulk search.
+        """
+        self._create_annotation(text=u'First one', usage_id='test-1')
+        self._create_annotation(text=u'Second note', usage_id='test-2')
+        self._create_annotation(text=u'Third note', usage_id='test-3')
+
+        def verify_bulk_search(usage_id):
+            """
+            Verify bulk search results based on usage id operation.
+
+            Arguments:
+                usage_id: String or List. The identifier string/list of the annotations XBlock(s).
+            """
+            if isinstance(usage_id, list):
+                results = self._get_search_results(usage_id=json.dumps(usage_id))
+                self.assertEqual(len(results), len(usage_id))
+                # Here we are reverse traversing the list because response contains annotation rows which are based on
+                # descending order of their update column.
+                for index, u_id in enumerate(usage_id[::-1]):
+                    self.assertEqual(results[index]['usage_id'], u_id)
+            else:
+                results = self._get_search_results(usage_id=usage_id)
+                self.assertEqual(len(results['rows']), 1)
+                self.assertEqual(results['rows'][0]['usage_id'], usage_id)
```

Comment:
I would check this with if part only without checking list. The response will be list in any case.